pulp_python/tests/functional/api/test_download_content.py

```diff
@@ -107,7 +107,45 @@ def test_basic_pulp_to_pulp_sync(self):
         ]
         unit_url = "/".join(url_fragments)
 
+        # Sync using old Pulp content api
         body["url"] = unit_url
         remote = self._create_remote(**body)
         repo2 = self._create_repo_and_sync_with_remote(remote)
         self.assertEqual(get_content_summary(repo2.to_dict()), PYTHON_LG_FIXTURE_SUMMARY)
+
+        # Sync using new PyPI endpoints
+        body["url"] = distro.base_url
+        remote = self._create_remote(**body)
+        repo3 = self._create_repo_and_sync_with_remote(remote)
+        self.assertEqual(get_content_summary(repo3.to_dict()), PYTHON_LG_FIXTURE_SUMMARY)
+
+    def test_full_fixtures_to_pulp_sync(self):
```

Comment:
I'm a little curious what happens if you sync an on-demand repo from pulp into pulp, in immediate mode.  Would be an interesting test to make sure nothing blows up.

Reply:
Although we might already have that test, I haven't looked.